Grand.Api/Infrastructure/DependencyEdmModel.cs

```diff
@@ -87,59 +89,87 @@ protected void RegisterProduct(ODataConventionModelBuilder builder)
             updateManufacturer.Returns<bool>();
 
             ActionConfiguration deleteManufacturer = product.Action("DeleteProductManufacturer");
-            deleteManufacturer.Parameter<string>("ManufacturerId").Required();
+            deleteManufacturer.Parameter<string>(nameof(ProductManufacturerDto.ManufacturerId)).Required();
             deleteManufacturer.Returns<bool>();
             #endregion
 
             //insert/update/delete picture
             #region Product picture
             ActionConfiguration createPicture = product.Action("CreateProductPicture");
-            createPicture.Parameter<ProductPictureDto>("productPicture");
+            createPicture.Parameter<string>(nameof(ProductPictureDto.PictureId)).Required();
+            createPicture.Parameter<string>(nameof(ProductPictureDto.MimeType)).Required();
+            createPicture.Parameter<string>(nameof(ProductPictureDto.SeoFilename)).Required();
+            createPicture.Parameter<string>(nameof(ProductPictureDto.AltAttribute)).Required();
+            createPicture.Parameter<int>(nameof(ProductPictureDto.DisplayOrder)).Required();
+            createPicture.Parameter<string>(nameof(ProductPictureDto.TitleAttribute)).Required();
             createPicture.Returns<bool>();
 
             ActionConfiguration updatePicture = product.Action("UpdateProductPicture");
-            updatePicture.Parameter<ProductPictureDto>("productPicture");
+            updatePicture.Parameter<string>(nameof(ProductPictureDto.PictureId)).Required();
+            updatePicture.Parameter<string>(nameof(ProductPictureDto.MimeType)).Required();
+            updatePicture.Parameter<string>(nameof(ProductPictureDto.SeoFilename)).Required();
+            updatePicture.Parameter<string>(nameof(ProductPictureDto.AltAttribute)).Required();
+            updatePicture.Parameter<int>(nameof(ProductPictureDto.DisplayOrder)).Required();
+            updatePicture.Parameter<string>(nameof(ProductPictureDto.TitleAttribute)).Required();
             updatePicture.Returns<bool>();
 
             ActionConfiguration deletePicture = product.Action("DeleteProductPicture");
-            deletePicture.Parameter<string>("PictureId").Required();
+            deletePicture.Parameter<string>(nameof(ProductPictureDto.PictureId)).Required();
             deletePicture.Returns<bool>();
             #endregion
 
             #region Product specification
             ActionConfiguration createSpecification = product.Action("CreateProductSpecification");
-            createSpecification.Parameter<ProductSpecificationAttributeDto>("productSpecification");
+            createSpecification.Parameter<string>(nameof(ProductSpecificationAttributeDto.Id)).Required();
+            createSpecification.Parameter<int>(nameof(ProductSpecificationAttributeDto.DisplayOrder));
+            createSpecification.Parameter<string>(nameof(ProductSpecificationAttributeDto.CustomValue)).Required();
```

Comment:
In my opinion it shouldn't be required, because it depends on the Attribute Type.

Reply:
You suggest to remove all required modifier or just this one?
Maybe I remove all except  "primarykey" like PicutreId, ManufacturerId etc?

Reply:
Just this one, because the custom values field is not always required.